### DISCUSSION TITLE
opt in to plain-text hover formatting

### DISCUSSION THREAD

**Discussion by jdtsmith** [joaotavora/eglot#1039] [Ideas]
Some LSP servers like pyright utilize HTML entities, escaping, etc. in their markdown-formatted response to hover requests, resulting in rather garbled display.  It would be useful to be able to configure eglot to **request plaintext only** for such servers, even if markdown-mode and its gfm-view-mode are installed.   Here's what it ends up looking like:

![image](https://user-images.githubusercontent.com/93749/190871667-f73cdc34-16a1-48f4-badd-ed5a8fd71cde.png)

compared to plaintext:

![image](https://user-images.githubusercontent.com/93749/190871701-31001f1c-fbf2-4109-9060-19b67635a44e.png)

**Comment by jdtsmith** [comment]
I'm happy to submit a PR if there is interest.  Particularly bad case [here](https://github.com/microsoft/pylance-release/issues/2231#issuecomment-1279616161).

**Reply by doolio** [reply]
Hey JD, did you ever put together a solution to this issue?

**Reply by jdtsmith** [reply]
I didn't, but it shouldn't be too bad. This [config](https://github.com/joaotavora/eglot/blob/e501275e06952889056268dabe08ccd0dbaf23e5/eglot.el#L736) basically enables markdown docs if they can be viewed in Emacs.  Just adding a new config variable to disable that on request would work.

**Reply by doolio** [reply]
OK thanks.

**Reply by d-miketa** [reply]
Thank you for diagnosing this issue, it's been bugging me for some time. Are you still open to submitting a PR with the proposed solution?

**Comment by joaotavora** [comment]
Yes, but submit it as a patch instead to Emacs's lisp/progmodes/eglot.el
and send it to ***@***.***  Make sure to CC: me (or better,
X-Debbugs-CC: me) in the email as I don't follow the bug-reporting list
regularly.

João

On Wed, Feb 8, 2023 at 12:40 PM Dom Miketa ***@***.***> wrote:

> Thank you for diagnosing this issue, it's been bugging me for some time.
> Are you still open to submitting a PR with the proposed solution?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/joaotavora/eglot/discussions/1039#discussioncomment-4904632>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAC6PQ32BXBHH53CD7ME5SDWWOH33ANCNFSM6AAAAAAQPDNU3M>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
João Távora

**Reply by doolio** [reply]
A related [discussion](https://github.com/jrblevin/markdown-mode/issues/661) from which one could argue that this issue would be best resolved in markdown-mode rather than Eglot?

**Comment by jdtsmith** [comment]
I submitted a simple patch that adds a new customization `eglot-prefer-plaintext`: bug#61373 (@joaotavora I don't have your email to copy).

**Reply by joaotavora** [reply]
It's in the file you edited.

**Reply by jdtsmith** [reply]
Copied you in.

**Comment by joaotavora** [comment]
Let's follow up in bug#61373

**Reply by LionyxML** [reply]
Hello there João (@joaotavora)!

Sorry to ressurect this topic and mess with your memory.

Just as a follow up (and maybe for some future one also getting into here). I followed the discussion on this bug, it led to two others, that let to other stuff and I got out of track.

So this feature actually moved to eldoc and won't be implemented by eglot, right?

Like, if I want to use Emacs and **nothing else** and get to stuff like this into eglot:

![Captura de Tela 2024-03-29 às 23 31 48](https://github.com/joaotavora/eglot/assets/16169950/39162c2f-08d1-4110-b0e8-d78fcbd558de)

Could you give me a hint on how to strip down stuff like ```?

I saw somewhere else that some servers like "R" can actually provide documentation text in other formats but for those that cannot I am finding hard to actually configure `eldoc-documentation-function.

Any help would me much appreciated, thanks.
